paddle/fluid/framework/tensor.cc

```diff
@@ -235,5 +237,13 @@ void Tensor::ResetHolderWithType(std::shared_ptr<memory::Allocation> holder,
 
 void Tensor::set_type(const proto::VarType::Type& type) { type_ = type; }
 
+#if defined(PADDLE_WITH_CUDA) || defined(PADDLE_WITH_HIP)
+void Tensor::RecordStream(const gpuStream_t& stream) {
```

Comment:
Adding backend/device specific interfaces to backend-agnostic data structures such as Tensor is strongly unrecommended, and will block works on related to Pten and Dygraph Refactor.

Taking "Tensor::RecordStream" as an example. When adding this interface to Tensor, the principle is that the interface has to "exist" whatsoever (regardless of the macro, the backend, the way you compile...), but you may use macros inside a source file to control how this interface is implemented towards different backends.

Reply:
The macros have been deleted, please check the new commit.